query-builder/src/main/java/com/datastax/oss/driver/api/querybuilder/relation/ColumnRelationBuilder.java

```diff
@@ -46,4 +46,26 @@ default ResultT contains(@NonNull Term term) {
   default ResultT containsKey(@NonNull Term term) {
     return build(" CONTAINS KEY ", term);
   }
+
+  /**
+   * Builds a NOT CONTAINS relation for the column.
+   *
+   * <p>Note that NOT CONTAINS support is only available in Cassandra 5.1 or later. See
+   * CASSANDRA-18584 for more information.
```

Comment:
For all the places we refer to CASSANDRA-18584 can we add a link (just so its clickable from javadocs)?

```suggestion
   * <a href="https://issues.apache.org/jira/browse/CASSANDRA-18584">CASSANDRA-18584</a> for more information.
```

Reply:
Since `NOT CONTAINS` and `NOT CONTAINS KEY` both unconditionally require `ALLOW FILTERING`.  It could be useful to mention that in the comments here as well.

Reply:
strike that, It seems that milestone 3 of the CEP aims to remove that requirement with SAI:
https://cwiki.apache.org/confluence/display/CASSANDRA/CEP-29%3A+CQL+NOT+operator#CEP29:CQLNOToperator-Milestone3

> Don't require ALLOW FILTERING if a negative restriction is used on an indexed column and the index supports efficient negative searches - skipping items that match the restricted term. This is useful in particular for implementing predicates of type `collection NOT CONTAINS element`, where collection is indexed.
To be implemented in SAI, after CEP-7 is merged. This can be implemented in SAI by introduction of "negating iterator" that would return a complement of a posting list. This way there is no need to change the implementation of the index data structure and such iterator would compose well with other SAI features, e.g. would support intersections and unions.

Reply:
A very definite +1 to the idea of updating all of the CASSANDRA-18584 references to hyperlinks... I like that quite a bit!

I wasn't super-worried about whether the CQL generated here works on a particular version of Cassandra (or, more accurately, a specific version of SAI) because this framework is only concerned with creating the actual CQL string.  There's not really any validation of any kind in this framework for confirming that what was created makes sense (or makes sense in all situations).  An extension of that for me is that the tests don't really make sense as examples of valid CQL; they're _only_ testing the ability of the query generator to create new strings.